C++/Morwenn's rewrite of Summer Dragonfly's GrailSort/grailsort.h

```diff
@@ -1186,8 +1189,8 @@ namespace grailsort_detail
                 return;
             }
 
-            BufferIterator extBuffer;
-            int extBufferLen;
+            BufferIterator extBuffer = 0;
```

Comment:
This will only work if the buffer is a contiguous container and its iterators are actual pointers. Using `BufferIterator extBuffer{};` might be more generic.

Reply:
> This will only work if the buffer is a contiguous container and its iterators are actual pointers. Using `BufferIterator extBuffer{};` might be more generic.

I haven't tested that change yet, I could try and see what will happen